### ISSUE TITLE
Add support for "inactive" properties

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot-proposals#4823]
### Describe the project you are working on

Godot etc

### Describe the problem or limitation you are having in your project

We have some properties that aren't always useful. Example is Button's `expand_icon` property. It's useless when you don't have icon. However it can't just be disabled/hidden, because you may want to configure the icon and then add it later.

Generally, there are 3 types of conditional properties:
- properties that will never be useful if their condition is not met. These can just be hidden. Examples are emission shape sizes in ParticlesMaterial - you only need to see the settings for currently selected shape
- properties that shouldn't be edited, but seeing their current value is useful. Examples are position and size of Controls inside Container
- properties that conditionally have no effect, but configuring them might be useful when the dependent value changes. Examples are `expand_icon` in Button, `region_rect` in Sprite2D and `stretch_margin` in TextureProgressBar

The 3rd case is currently not supported properly, i.e. there is a debate whether these properties should be hidden or not, but there is no middle ground, which would be the best in this case.
Some relevant links: https://github.com/godotengine/godot/pull/62423 / https://github.com/godotengine/godot-proposals/discussions/4755

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Aside from hiding or disabling properties, we could have a third option - making them inactive. This would be a bit similar to configuration warnings.
![image](https://user-images.githubusercontent.com/2223172/177541546-1e2c80c8-4044-4ade-9573-ebd445a49365.png)
When property is inactive, its name would be greyed-out (or have some tint) and the description would have a warning-like message saying why is it inactive.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Probably add a new property hint `PROEPRTY_HINT_INACTIVE`, with the optional warning message stored in hint text. This would be alternate option for "disabling" properties in `_validate_property()`.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Nah, no way.

### Is there a reason why this should be core and not an add-on in the asset library?

It's core feature.

**Comment by YuriSizov** [comment]
> Probably add a new property hint `PROEPRTY_HINT_INACTIVE`, with the optional warning message stored in hint text.

I assume properties would still have their base hint in place, and so the hint string would already be taken a lot of time. And it should really be a usage flag, not a hint flag. Which brings me to...

We already have `PROPERTY_USAGE_READ_ONLY`. 👀 It was used in the control's layout properties before the rework to prevent editing properties which are controlled by containers (or vice versa, are useless outside of containers). It is still in use in some other nodes.

To implement warnings instead I propose a new method like `_get_configuration_warning()` but for properties, `_get_property_warning(String property_name)`. It can be executed in a similar manner to `_validate_property()` to update the warnings. Those warnings would be added just like you've shown, as an addendum in their tooltips.

> properties that shouldn't be edited, but seeing their current value is useful. Examples are position and size of Controls inside Container

This won't immediately benefit from the inactive/readonly properties. The problem with this particular PR is that we have a guided layout for properties to prevent common mistakes people make in 3.x. We were asked to show all properties in the discussion preceding this change, but we decided against it. So to bring readonly properties back it would require some thinking on how to keep the layout UI guided while giving more information to power users.

That said, the main issue is probably with the remote inspector, not the local one. And a likely solution is to still show `PROPERTY_USAGE_NO_EDITOR` props in the remote inspector during the debugging session.

**Comment by KoBeWi** [comment]
> If your intention is to keep them editable and not completely readonly, then the name inactive is misleading.

Not really, "inactive" i.e. "ineffective" - you can set it, but it has no effect.

> To implement warnings instead of hint text I propose a new method like _get_configuration_warning() but for properties, _get_property_warning(String property_name). It can be executed in a similar manner to _validate_property() to update the warnings.

Good idea 👀

**Comment by Calinou** [comment]
Can this system be used for properties that require a specific rendering method, but the rendering method in question isn't currently being used? cc @BastiaanOlij
